x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceNamedWriteablesProvider.java

```diff
@@ -27,8 +30,11 @@ private InferenceNamedWriteablesProvider() {}
     public static List<NamedWriteableRegistry.Entry> getNamedWriteables() {
         List<NamedWriteableRegistry.Entry> namedWriteables = new ArrayList<>();
 
-        // Empty default settings
-        namedWriteables.add(new NamedWriteableRegistry.Entry(EmptyTaskSettings.class, EmptyTaskSettings.NAME, EmptyTaskSettings::new));
+        // Empty default task settings
+        namedWriteables.add(new NamedWriteableRegistry.Entry(TaskSettings.class, EmptyTaskSettings.NAME, EmptyTaskSettings::new));
```

Comment:
Is this a breaking change since we're switching from the concrete class to the interface? If so, I doubt anyone has been using this in serverless because we don't have the elser model released for it.